source/code/plugins/mysql_workload_lib.rb

```diff
@@ -0,0 +1,442 @@
+      #   +--------------------+--------+-------------------------------+
+      # Note that Size may be NULL if database has no tables
+
+    else
+      # Number of tables in given database and the table size in bytes
+      @query_get_database = 'select b.schema_name as "DatabaseName",COUNT(a.table_name) as "NumberOfTables",
+                                                                       SUM(a.data_length) + SUM(a.index_length) as "SizeInBytes"
+                                                                         from information_schema.schemata b
+                                                                         left join information_schema.tables a on b.schema_name = a.table_schema
+                                                                         where b.schema_name =\'' + @database + '\' group by b.schema_name'
+    end
+  end
+
+  def get_connection
+    # Opens a connection to mysql server instance
+    begin
+      return Mysql2::Client.new({
+        :host => @host,
+        :port => @port,
+        :username => @username,
+        :password => @password,
+        :database => @database,
+        :encoding => @encoding,
+        :reconnect => true
+      })
+    rescue Exception => e
+      @log.warn_once("mysql_workload: #{e}")
+    end
+  end
+
+  def query(input_query)
+    if @mock_interface.nil?
+      @mysql ||= get_connection
+      begin
+        return @mysql.query(input_query, :cast => false, :cache_rows => false)
+      rescue Exception => e
+        @log.warn_once("mysql_workload: #{e}")
+      end
+    else
+      return @mock_interface.query(input_query)
+    end
+
+  end
+
+  def get_mysql_hostname
+    query("SHOW VARIABLES LIKE 'hostname'").each do |row|
+      return row.fetch('Value')
+    end
+  end
+
+  def transform_row(row)
+    # creates a new hash table where keys from the input row are replaced with the Variable_name key name, and the value of the key Variable_name
+    # Example {VariableName => connectedthreads, Value => 20} will be replaced with {connectedthreads = > 20}
+    result = Hash.new()
+    result.store(row["Variable_name"], row["Value"] )
+    return result
+  end
+
+  def get_value (input_array, key_to_find)
+    # finds given key in the array of hash tables and returns it's value
+    value = -1
+    # retrieves the entry value from the array of hash tables based on the given key
+    input_array.each do |row|
+      if (row.has_key?(key_to_find))
+        value = row.fetch(key_to_find)
+        break
+      end
+    end
+    return value
+  end
+
+  def get_server_database_type
+    #retrieve engine info from default_storage_engine
+    engine_query = "show variables like 'default_storage_engine'" #some versions of mysql might not populated this field so use default
+    engine_info = query(engine_query)
+    engine_info_value = "InnoDB" # default
+    engine_info.each do |row|
+      engine_info_value = row["Value"]
+    end
+    return engine_info_value
```

Comment:
It is actually always only one value. Each loop here is not needed at all. Will remove the each.